h2o-logger/src/main/java/water/logging/LoggerFactory.java

```diff
@@ -0,0 +1,56 @@
+package water.logging;
+
+
+
+public class LoggerFactory {
+    private static final String DEFAULT_CLASS_TO_CHECK = "org.slf4j.LoggerFactory";
+    private static final LoggerFactory INSTANCE = new LoggerFactory(DEFAULT_CLASS_TO_CHECK);
+    
+    private final String slf4jClassName;
+    private final boolean isSlf4JAvailable; 
+
+    LoggerFactory(String slf4jClass) {
+        slf4jClassName = slf4jClass;
+        isSlf4JAvailable = isSlf4JAvailable();
+    }
+    
+    /**
+     * Returns new logger for each invocation. Logger logs to console / SLF4J (custom wrapper around SLF4J) depending on whether SLF4J is available.
+     *
+     * @param clazz class from which getLogger() is called
+     * @return Slf4JLogger if SLF4J is on the classpath, ConsoleLogger if not
+     */
+    public static Logger getLogger(Class<?> clazz) {
+        return INSTANCE.getCustomLogger(clazz);
+    }
+
+    
+    /**
+     * Returns new logger for each invocation.  Logger logs to console / SLF4J (custom wrapper around SLF4J) depending on whether SLF4J is available.
+     *
+     * @param clazz class from which getLogger() is called
+     * @return Slf4JLogger if SLF4J is on the classpath, ConsoleLogger if not
+     */
+    public Logger getCustomLogger(Class<?> clazz) {
+        if (isSlf4JAvailable) {
```

Comment:
can we make the decision just once?

IMHO better for the MOJO use case

Reply:
this is done once per class static-init so one if() is not a big issue I think